### PR TITLE
Upgrade Go and go-github

### PR DESCRIPTION
**Pull request by iul1an**
### Description of your changes

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: internal/util/util.go [hunk 1]
```diff
@@ -276,11 +276,3 @@ func Int64DerefToPointer(ptr *int64, def int64) *int64 {
 	i := pointer.Int64Deref(ptr, def)
 	return &i
 }
-
-// BoolToInt converts a boolean value to an integer
-func BoolToInt(b bool) int {
```
Comment (kaessert) [inline internal/util/util.go:281]: This is used here :) https://github.com/kaessert/provider-github/blob/6179da8d5b7e94e2891afeb93b0d32ad0d00cdf2/internal/controller/repository/repository.go#L808

Comment (iul1an) [reply internal/util/util.go:281]: It's not used anymore: https://github.com/kaessert/provider-github/pull/20/files#diff-a961e88c47a586d25a55b6104120a8fbf3180116aa0ede211c60b2d05f614648L808

Comment (kaessert) [reply internal/util/util.go:281]: I see, my bad :)

**Review by kaessert — APPROVED** [review]

**Comment by kaessert** [comment]
Thanks again for your contribution @iul1an !

**Pull request merged**